internal/controller/entra/custom_mail.go

```diff
@@ -1,45 +1,81 @@
 package entra
 
 import (
+	"bytes"
 	"context"
+	"html/template"
+
+	"os"
 
 	graphmodels "github.com/microsoftgraph/msgraph-sdk-go/models"
 	graphusers "github.com/microsoftgraph/msgraph-sdk-go/users"
 	"github.com/ucl-arc-tre/portal/internal/config"
+	openapi "github.com/ucl-arc-tre/portal/internal/openapi/web"
 	"github.com/ucl-arc-tre/portal/internal/types"
 )
 
-func (c *Controller) SendCustomInviteNotification(ctx context.Context, email string, sponsor types.Sponsor) error {
-	// use graph to send email saying so-and-so has invited you to the portal
+func (c *Controller) createCustomEmail(ctx context.Context, subject string, emails []string, content string) error {
+	// set up the email
 
 	requestBody := graphusers.NewItemSendMailPostRequestBody()
 	message := graphmodels.NewMessage()
-	subject := "Notification: You have been invited to the UCL ARC Services Portal"
-	message.SetSubject(&subject)
+	message.SetSubject(&subject) // set from args
 	body := graphmodels.NewItemBody()
 	contentType := graphmodels.HTML_BODYTYPE
 	body.SetContentType(&contentType)
 
-	content := ""
-	if sponsor.ChosenName != "" {
-		content = "You have been invited to join the UCL ARC Services Portal by " + string(sponsor.ChosenName)
-	} else {
-		content = "You have been invited to join the UCL ARC Services Portal by " + string(sponsor.Username)
+	t, err := template.ParseFiles("internal/controller/entra/base_mail_template.html")
+	if err != nil {
+		return err
 	}
-	content = content + "<br><br>You can sign into the portal at <a href='https://portal.arc.ucl.ac.uk'>https://portal.arc.ucl.ac.uk</a> <br><br> This is an automated message, but this mailbox is monitored so you can contact us if you need assistance."
 
-	body.SetContent(&content)
-	message.SetBody(body)
+	templateReader := new(bytes.Buffer)
+	err = t.Execute(templateReader, struct {
+		Content string
+	}{
+		Content: content, // set the content from the args
+	})
+	if err != nil {
+		return err
+	}
+	htmlContent := templateReader.String()
+
+	body.SetContent(&htmlContent)
 
-	recipient := graphmodels.NewRecipient()
-	emailAddress := graphmodels.NewEmailAddress()
-	emailAddress.SetAddress(&email)
-	recipient.SetEmailAddress(emailAddress)
+	// has attachments to allow banner to work
+	hasAttachments := true
+	message.SetHasAttachments(&hasAttachments)
+	banner := graphmodels.NewFileAttachment()
+	oDataType := "#microsoft.graph.fileAttachment"
+	banner.SetOdataType(&oDataType)
+	svgType := "image/svg+xml"
+	banner.SetContentType(&svgType)
+	svgName := "ucl-banner.svg"
+	banner.SetName(&svgName)
+	svgId := "uclBanner"
+	banner.SetContentId(&svgId)
+	banner.SetIsInline(&hasAttachments)
 
-	toRecipients := []graphmodels.Recipientable{
-		recipient,
+	// read the file into bytes
+	svgFile, err := os.ReadFile("internal/controller/entra/ucl-banner.svg")
+	if err != nil {
+		return err
 	}
```

Comment:
I'd be tempted to embed this cf. https://github.com/ucl-arc-tre/portal/blob/17b8bf02634e5cf693a7164d84830bd011470efe/internal/service/agreements/init.go#L13-L14